web/search.dart

```diff
@@ -90,56 +87,59 @@ class _Search {
   final Index index;
   final Uri uri;
 
-  late final listBox = document.createElement('div')
+  late final listBox = document.createElement('div') as HTMLElement
     ..setAttribute('role', 'listbox')
     ..setAttribute('aria-expanded', 'false')
     ..style.display = 'none'
-    ..classes.add('tt-menu')
-    ..append(moreResults)
-    ..append(searchResults);
+    ..classList.add('tt-menu')
+    ..appendChild(moreResults)
+    ..appendChild(searchResults);
 
   /// Element used in [listBox] to inform the functionality of hitting enter in
   /// search box.
   late final moreResults = document.createElement('div')
-    ..classes.add('enter-search-message');
+    ..classList.add('enter-search-message');
 
   /// Element that contains the search suggestions in a new format.
   late final searchResults = document.createElement('div')
-    ..classes.add('tt-search-results');
+    ..classList.add('tt-search-results');
 
   String? storedValue;
   String actualValue = '';
-  final List<Element> suggestionElements = <Element>[];
+  final List<HTMLElement> suggestionElements = <HTMLElement>[];
   List<IndexItem> suggestionsInfo = <IndexItem>[];
   int selectedElement = -1;
 
   _Search(this.index) : uri = Uri.parse(window.location.href);
 
-  void initialize(InputElement inputElement) {
+  void initialize(HTMLInputElement inputElement) {
     inputElement.disabled = false;
     inputElement.setAttribute('placeholder', 'Search API Docs');
     // Handle grabbing focus when the user types '/' outside of the input.
-    document.addEventListener('keydown', (Event event) {
-      if (event is! KeyboardEvent) {
-        return;
-      }
-      if (event.key == '/' && document.activeElement is! InputElement) {
-        event.preventDefault();
-        inputElement.focus();
-      }
-    });
+    document.addEventListener(
+      'keydown',
+      (Event event) {
+        if (event is! KeyboardEvent) {
+          return;
+        }
+        if (event.key == '/' && document.activeElement is! HTMLInputElement) {
+          event.preventDefault();
+          inputElement.focus();
+        }
+      }.toJS,
+    );
 
     // Prepare elements.
-    var wrapper = document.createElement('div')..classes.add('tt-wrapper');
+    var wrapper = document.createElement('div')..classList.add('tt-wrapper');
     inputElement
-      ..replaceWith(wrapper)
+      ..replaceWith(wrapper as JSAny) // TODO(parlough): How to without cast?
```

Comment:
@srujzs Is there any better way to use [`replaceWith`](https://pub.dev/documentation/web/latest/web/ElementExtension/replaceWith.html) and its related methods that accept `JSAny nodes`? I'd expect to be able to just specify an `Element` or at least a array/list of `Element`.

Reply:
This is temporary. :) Since `@staticInterop` types (because they're classes) can't implement extension types, we had to remove the `implements` clauses. Once we release a version where all of these classes move to extension types themselves, the `implements` clauses will go back on and a cast will not be needed.